### ISSUE TITLE
[Web]TextformField shows error when trying to type with keyboard

### ISSUE THREAD

**Issue by Macacoazul01** [flutter/flutter#75076]
Since some flutter upgrade, i can't use any TextformField with a controller anymore (and without it too).

Everytime i try to type something, vs code gives me this error on the debug console:
```
Error: Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart:923:10
zone != null
is not true
    at Object.throw_ [as throw] (http://localhost:51528/dart_sdk.js:5348:11)
    at Object.assertFailed (http://localhost:51528/dart_sdk.js:5284:15)
    at Object.invoke (http://localhost:51528/dart_sdk.js:184390:31)
    at _engine.EnginePlatformDispatcher.__.invokeOnKeyData (http://localhost:51528/dart_sdk.js:164864:15)
    at _engine.KeyboardBinding.__.[_onKeyData] (http://localhost:51528/dart_sdk.js:164031:49)
    at _engine.KeyboardConverter.new.handleEvent (http://localhost:51528/dart_sdk.js:164400:52)
    at http://localhost:51528/dart_sdk.js:164040:33
    at http://localhost:51528/dart_sdk.js:164013:18
```

my flutter doctor:

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, 1.26.0-18.0.pre.90, on Microsoft Windows [versÃ£o 10.0.19042.746], locale pt-BR)
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code (version 1.52.1)
[√] Connected device (2 available)
```

Simple code trying to test the textformfield:

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  TextEditingController _textControllerNome=TextEditingController(text: "");

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            TextFormField(controller: _textControllerNome,),
            TextFormField()
          ],
        ),
      ),// This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```
</details>

Does someone knows any version that i can downgrade to, so the i'll be able to run my web app again?

**Comment by Macacoazul01** [comment]
On beta this error doesn't exist.

"""Solved""" with:

```
flutter channel beta

flutter upgrade
```

**Comment by iapicca** [comment]
@Macacoazul01 
I actually I care repro the error with the simple counter app
- run 
```console
flutter  create issue
cd issue
flutter run -d chrome
```
- tap the fab
- error in logs
```bash
Error: Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart:923:10
zone != null
is not true
    at Object.throw_ [as throw] (http://localhost:37691/dart_sdk.js:5348:11)
    at Object.assertFailed (http://localhost:37691/dart_sdk.js:5284:15)
    at Object.invoke (http://localhost:37691/dart_sdk.js:184390:31)
    at _engine.EnginePlatformDispatcher.__.invokeOnKeyData (http://localhost:37691/dart_sdk.js:164864:15)
    at _engine.KeyboardBinding.__.[_onKeyData] (http://localhost:37691/dart_sdk.js:164031:49)
    at _engine.KeyboardConverter.new.handleEvent (http://localhost:37691/dart_sdk.js:164400:52)
    at http://localhost:37691/dart_sdk.js:164040:33
    at http://localhost:37691/dart_sdk.js:164013:18

```

```console
[✓] Flutter (Channel master, 1.26.0-18.0.pre.90, on Linux, locale en_US.UTF-8)
    • Flutter version 1.26.0-18.0.pre.90 at /home/francesco/snap/flutter/common/flutter
    • Framework revision 25134a16d1 (32 hours ago), 2021-01-29 23:52:30 -0800
    • Engine revision ce14c8a841
    • Dart version 2.12.0 (build 2.12.0-276.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/francesco/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_SDK_ROOT = /home/francesco/Android/Sdk
    • Java binary at: /usr/lib/jvm/java-1.8.0-openjdk-amd64/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_275-8u275-b01-0ubuntu1~20.04-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /snap/bin/chromium

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Linux
    • Chrome (web)    • chrome • web-javascript • Chromium 88.0.4324.96 snap

! Doctor found issues in 1 category.

```

**Comment by lfwells** [comment]
Have the same issue here after an upgrade (note this is a major upgrade from over 10 months ago, so I assume I've missed something that I need to change about my project)

`Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, 1.26.0-18.0.pre.90, on Microsoft Windows [Version
    10.0.19042.746], locale en-AU)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code (version 1.52.1)      
[√] Connected device (1 available)`

**Comment by Macacoazul01** [comment]
It was everything ok with the master branch ~5 days ago, but them this error started happening. Seems like something related with the null safety migration (?)

**Comment by kunit1** [comment]
This exception is happening for me on the flutter web CanvasKit version on master.

**Comment by pedromassangocode** [comment]
Easily reproducible on master channel.

```console
[ +140 ms] Error: Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart:923:10
           zone != null
           is not true
               at Object.throw_ [as throw] (http://localhost:53475/dart_sdk.js:5348:11)
               at Object.assertFailed (http://localhost:53475/dart_sdk.js:5284:15)
               at Object.invoke (http://localhost:53475/dart_sdk.js:184390:31)
               at _engine.EnginePlatformDispatcher.__.invokeOnKeyData (http://localhost:53475/dart_sdk.js:164864:15)
               at _engine.KeyboardBinding.__.[_onKeyData] (http://localhost:53475/dart_sdk.js:164031:49)
               at _engine.KeyboardConverter.new.handleEvent (http://localhost:53475/dart_sdk.js:164400:52)
               at http://localhost:53475/dart_sdk.js:164044:33
               at http://localhost:53475/dart_sdk.js:164013:18
```

<details>
<summary>logs</summary>

```bash
[        ] Synced 29.4MB.
[        ] <- accept
[        ] Caching compiled dill
[ +341 ms] Using Google Chrome 88.0.4324.96 

[+1863 ms] [CHROME]:
[   +1 ms] [CHROME]:DevTools listening on ws://127.0.0.1:53524/devtools/browser/84cac4a1-9f4c-4fc2-a875-048dade02a6b
[+1317 ms] DwdsInjector: Received request for entrypoint at http://localhost:53475/main_module.bootstrap.js
[   +9 ms] MetadataProvider: Loading debug metadata...
[  +51 ms] MetadataProvider: Loaded debug metadata (no sound null safety)
[   +9 ms] DwdsInjector: Injected debugging metadata for entrypoint at http://localhost:53475/main_module.bootstrap.js
[+3000 ms] ChromeProxyService: Initializing expression compiler for main_module.bootstrap.js with sound null safety: false
[ +704 ms] DevHandler: Debug service listening on ws://127.0.0.1:53597/hVaPEO89_vs=/ws

[  +12 ms] Debug service listening on ws://127.0.0.1:53597/hVaPEO89_vs=/ws
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[   +1 ms] 🔥  To hot restart changes while running, press "r" or "R".
[        ] For a more detailed help message, press "h". To quit, press "q".
[+51458 ms] Error: Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart:923:10
            zone != null
            is not true
                at Object.throw_ [as throw] (http://localhost:53475/dart_sdk.js:5348:11)
                at Object.assertFailed (http://localhost:53475/dart_sdk.js:5284:15)
                at Object.invoke (http://localhost:53475/dart_sdk.js:184390:31)
                at _engine.EnginePlatformDispatcher.__.invokeOnKeyData (http://localhost:53475/dart_sdk.js:164864:15)
                at _engine.KeyboardBinding.__.[_onKeyData] (http://localhost:53475/dart_sdk.js:164031:49)
                at _engine.KeyboardConverter.new.handleEvent (http://localhost:53475/dart_sdk.js:164400:52)
                at http://localhost:53475/dart_sdk.js:164040:33
                at http://localhost:53475/dart_sdk.js:164013:18
[ +102 ms] Error: Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart:923:10
           zone != null
           is not true
               at Object.throw_ [as throw] (http://localhost:53475/dart_sdk.js:5348:11)
               at Object.assertFailed (http://localhost:53475/dart_sdk.js:5284:15)
               at Object.invoke (http://localhost:53475/dart_sdk.js:184390:31)
               at _engine.EnginePlatformDispatcher.__.invokeOnKeyData (http://localhost:53475/dart_sdk.js:164864:15)
               at _engine.KeyboardBinding.__.[_onKeyData] (http://localhost:53475/dart_sdk.js:164031:49)
               at _engine.KeyboardConverter.new.handleEvent (http://localhost:53475/dart_sdk.js:164400:52)
               at http://localhost:53475/dart_sdk.js:164044:33
               at http://localhost:53475/dart_sdk.js:164013:18
[  +66 ms] Error: Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart:923:10
           zone != null
           is not true
               at Object.throw_ [as throw] (http://localhost:53475/dart_sdk.js:5348:11)
               at Object.assertFailed (http://localhost:53475/dart_sdk.js:5284:15)
               at Object.invoke (http://localhost:53475/dart_sdk.js:184390:31)
               at _engine.EnginePlatformDispatcher.__.invokeOnKeyData (http://localhost:53475/dart_sdk.js:164864:15)
               at _engine.KeyboardBinding.__.[_onKeyData] (http://localhost:53475/dart_sdk.js:164031:49)
               at _engine.KeyboardConverter.new.handleEvent (http://localhost:53475/dart_sdk.js:164400:52)
               at http://localhost:53475/dart_sdk.js:164040:33
               at http://localhost:53475/dart_sdk.js:164013:18
[ +140 ms] Error: Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart:923:10
           zone != null
           is not true
               at Object.throw_ [as throw] (http://localhost:53475/dart_sdk.js:5348:11)
               at Object.assertFailed (http://localhost:53475/dart_sdk.js:5284:15)
               at Object.invoke (http://localhost:53475/dart_sdk.js:184390:31)
               at _engine.EnginePlatformDispatcher.__.invokeOnKeyData (http://localhost:53475/dart_sdk.js:164864:15)
               at _engine.KeyboardBinding.__.[_onKeyData] (http://localhost:53475/dart_sdk.js:164031:49)
               at _engine.KeyboardConverter.new.handleEvent (http://localhost:53475/dart_sdk.js:164400:52)
               at http://localhost:53475/dart_sdk.js:164044:33
               at http://localhost:53475/dart_sdk.js:164013:18
```
</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel master, 1.26.0-18.0.pre.90, on macOS 11.1 20C69 darwin-x64, locale en)
    • Flutter version 1.26.0-18.0.pre.90 at /Users/pedromassango/Code/flutter_master
    • Framework revision 25134a16d1 (2 days ago), 2021-01-29 23:52:30 -0800
    • Engine revision ce14c8a841
    • Dart version 2.12.0 (build 2.12.0-276.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 52.2.5
    • Dart plugin version 203.6912

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 88.0.4324.96

! Doctor found issues in 1 category.
```

</details>

**Comment by iapicca** [comment]
@pedromassangocode 
I think the problem is deeper than text input
as mentioned [above](https://github.com/flutter/flutter/issues/75076#issuecomment-770406986) it occurs with a simple counter app

**Comment by pedromassangocode** [comment]
Hi @iapicca 
I cannot reproduce with the counter app on MacOS.

![gif](https://user-images.githubusercontent.com/67424416/106449491-50439880-6484-11eb-86c3-c391b3f7bca8.gif)


<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {

    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```
</details>


<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel master, 1.26.0-18.0.pre.90, on macOS 11.1 20C69 darwin-x64, locale en)
    • Flutter version 1.26.0-18.0.pre.90 at /Users/pedromassango/Code/flutter_master
    • Framework revision 25134a16d1 (2 days ago), 2021-01-29 23:52:30 -0800
    • Engine revision ce14c8a841
    • Dart version 2.12.0 (build 2.12.0-276.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 52.2.5
    • Dart plugin version 203.6912

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 88.0.4324.96

! Doctor found issues in 1 category.
```

</details>

**Comment by Aerofluxx** [comment]
Reproduce with `1.26.0-18.0.pre.90` and also with `1.26.0-18.0.pre.91` in FlutterWeb.

```
Error: Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart:923:10
zone != null
is not true
    at Object.throw_ [as throw] (http://localhost:62543/dart_sdk.js:5348:11)
    at Object.assertFailed (http://localhost:62543/dart_sdk.js:5284:15)
    at Object.invoke (http://localhost:62543/dart_sdk.js:184390:31)
    at _engine.EnginePlatformDispatcher.__.invokeOnKeyData (http://localhost:62543/dart_sdk.js:164864:15)
    at _engine.KeyboardBinding.__.[_onKeyData] (http://localhost:62543/dart_sdk.js:164031:49)
    at _engine.KeyboardConverter.new.handleEvent (http://localhost:62543/dart_sdk.js:164400:52)
    at http://localhost:62543/dart_sdk.js:164044:33
    at http://localhost:62543/dart_sdk.js:164013:18
```
<p>
<details>
<summary>flutter doctor -v</summary>

<p>

```
[✓] Flutter (Channel master, 1.26.0-18.0.pre.91, on macOS 11.1 20C69 darwin-arm, locale de-AT)
    • Flutter version 1.26.0-18.0.pre.91 at /Users/mike/flutter
    • Framework revision cb14df5659 (10 minutes ago), 2021-02-01 06:17:38 -0500
    • Engine revision ce14c8a841
    • Dart version 2.12.0 (build 2.12.0-276.0.dev)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/mike/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = /Users/mike/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 52.1.5
    • Dart plugin version 203.6803

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.19.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 11.1 20C69 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 88.0.4324.96

• No issues found!

```
</details>

**Comment by Pitazzo** [comment]
Same problem here

**Comment by iapicca** [comment]
@Pitazzo 
the PR has been merged few hours ago, make sure you are using the latest master

**Comment by Pitazzo** [comment]
@iapicca maybe I'm behind a cache or something, but I just run `flutter upgrade` and the bug is still here :/

**Comment by kunit1** [comment]
@Pitazzo The engine PR still has not rolled into the framework. I would say try upgrading in a few hours.

**Comment by Macacoazul01** [comment]
still not rolled on 1.26.0-18.0.pre.157
@Pitazzo try using flutter channel beta while the fix isnt avaliable on master

**Comment by mdebbar** [comment]
The fix landed in flutter master around 3 hours ago: https://github.com/flutter/flutter/pull/75309

**Comment by Macacoazul01** [comment]
Its fixed on master 1.26.0-18.0.pre.169
